Feedback on build documentation

Trying to build from the docs got me some figuring out, so I propose we update the docs to include the other stuff. But before I submit a docs change, I'd like to know what of those "figuring out" things are "commonly known", assumed or just me not getting it right.

So, here's a list of things I had to do in addition to the base docs:
1. After installing `conda`, I have created a script that enters/leaves the virtualenv, as I don't want to enter it on login (its default behaviour is to add it to `.bashrc`, which is silly). Basically, I _moved_ what `conda` added to `.bashrc` into a new script, and added two new aliases: `alias enter-plaidml='source plaidml-env-script'` and `alias leave-plaidml='conda deactivate'`.
2. To allow OpenMP testing, PlaidML needs `llvm-lit`, `FileCheck` and `not` on the PATH, but the `llvm-project-build` directory didn't have it. I have built LLVM+Clang+MLIR on my box and copied the files to a directory in the PATH (same as the env script). I don't know how that directory is pulled from, but it would be nice to have them there and maybe even automatically add it to the PATH.
3. I created an Ubuntu 22.04 container, and when running `./configure` I was told there's no support for OpenCV on that distro (even though I can install its packages, so I'm confused). The interim solution is to run `./configure --no_openvino`.
4. The docs don't mention submodules, but the two directories under `vendor` (`mlir-hlo` and `openvino`) don't get initialised by default. I suggest adding a step to `git submodule update --init --recursive` before calling `./configure`.
5. I had no `ccache` on my container, but the build system still chose to use it, then I got a `ccache: not found` when building. I just installed `ccache` (and `clang` and `lld` for good measure), but it would be good to only use it when it's on the PATH. Installing `clang` and adding `export CC=clang CXX=clang++` didn't make `configure` choose it for the build.

After the CMake is done, I get a nice message on how to run the unit tests, but the build has many error messages like:

```
../../plaidml/core/core.h:93:5: error: 'fn' cannot be used as a function
../../plaidml/core/core.h: In instantiation of 'void plaidml::ffi::call_void(F, Args ...) [with F = int; Args = {void (*)(plaidml_error*, plaidml_expr*, long unsigned int, plaidml_dim_expr**), plaidml_expr*, long unsigned int, plaidml_dim_expr**}]':
```

```
../../plaidml/op/lib/ops.cc: In function 'plaidml::edsl::Value plaidml::op::lib::variance(const plaidml::edsl::Value&)':
../../plaidml/op/lib/ops.cc:3629:33: error: no matching function for call to 'plaidml::edsl::Value::Value(<brace-enclosed initializer list>)'
 3629 |   return Value{SumSqDiff / denom};
      |                                 ^
```

```
../../plaidml/op/lib/ops.cc:3338:17: error: no matching function for call to 'plaidml::edsl::TensorDim::TensorDim()'
 3338 |       TensorDim G, C, K;
      |                 ^
```

```
/usr/include/c++/11/type_traits:955:52: error: static assertion failed: template argument must be a complete class or an unbounded array
  955 |       static_assert(std::__is_complete_or_unbounded(__type_identity<_Tp>{}),
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
```

IIUC, all errors have plenty of candidates, which means the compiler's ability to guess the types is going wrong. This is using `ccache c++` which defaults to `gcc`, and the version on my system is 11.2.0. I found no way to change it to clang.

I think the docs should mention how to create the environment that is known to work (ex. Ubuntu 20.04), so that people with newer OSs don't try to chase weird build errors. Creating a docker that has that environment would be a nice touch, too.

--- a/configure
+++ b/configure
@@ -114,6 +114,23 @@ class Configure:
             sys.exit(1)
         run([pre_commit, 'install'], check=True)
 
+    def configure_compiler(self, args):
+        cmake_compiler_flags = []
+
+        cc = args.cc
+        if cc is None:
+            cc = os.getenv('CC')
+        cxx = args.cxx
+        if (cxx is None):
+            cxx = os.getenv('CXX')
+
+        if cc:
+            cmake_compiler_flags.append(f'-DCMAKE_C_COMPILER={cc}')
+        if cxx:
+            cmake_compiler_flags.append(f'-DCMAKE_CXX_COMPILER={cxx}')
+
+        return cmake_compiler_flags
+
     def configure_cmake(self, args):
         build_dir = pathlib.Path(f'build-{args.target}').resolve() / args.type
         cmd = [
@@ -135,6 +152,7 @@ class Configure:
             cmd.append('-DLOCAL_LLVM_DIR={}'.format(args.local_llvm))
         else:
             cmd.append('-ULOCAL_LLVM_DIR')
+        cmd.extend(self.configure_compiler(args))
         run(cmd, check=True)
 
 
@@ -159,6 +177,12 @@ def main():
     parser.add_argument('--temp',
                         type=pathlib.Path,
                         help='Configures the temporary directory for cached assets.')
+    parser.add_argument('--cc',
+                        type=pathlib.Path,
+                        help='Path/name of the C compiler.')
+    parser.add_argument('--cxx',
+                        type=pathlib.Path,
+                        help='Path/name of the C++ compiler.')
     args = parser.parse_args()
 
     print("Configuring PlaidML build environment")

